src/state/slices/opportunitiesSlice/opportunitiesSlice.ts

```diff
@@ -84,35 +83,37 @@ export const opportunities = createSlice({
     clear: () => initialState,
     upsertOpportunityMetadata: {
       reducer: (draftState, { payload }: { payload: GetOpportunityMetadataOutput }) => {
-        const payloadIds = Object.keys(payload.byId) as OpportunityId[]
-
         draftState[payload.type].byId = Object.assign(
           {},
           draftState[payload.type].byId,
           payload.byId,
         )
-        draftState[payload.type].ids = uniq([...draftState[payload.type].ids, ...payloadIds])
+        draftState[payload.type].ids = Object.keys(draftState[payload.type].byId)
       },
       // Use the `prepareAutoBatched` utility to automatically
       // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
       prepare: prepareAutoBatched<GetOpportunityMetadataOutput>(),
     },
-    upsertOpportunityAccounts: (
-      draftState,
-      { payload }: { payload: GetOpportunityUserDataOutput },
-    ) => {
-      draftState[payload.type].byAccountId = merge(
-        draftState[payload.type].byAccountId,
-        payload.byAccountId,
-      )
+    upsertOpportunityAccounts: {
+      reducer: (draftState, { payload }: { payload: GetOpportunityUserDataOutput }) => {
+        draftState[payload.type].byAccountId = merge(
+          draftState[payload.type].byAccountId,
+          payload.byAccountId,
+        )
+      },
+      // Use the `prepareAutoBatched` utility to automatically
+      // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
+      prepare: prepareAutoBatched<GetOpportunityUserDataOutput>(),
     },
-    upsertUserStakingOpportunities: (
-      draftState,
-      { payload }: { payload: GetOpportunityUserStakingDataOutput },
-    ) => {
-      const payloadIds = Object.keys(payload.byId) as UserStakingId[]
-      draftState.userStaking.byId = merge(draftState.userStaking.byId, payload.byId)
-      draftState.userStaking.ids = uniq([...draftState.userStaking.ids, ...payloadIds])
+    upsertUserStakingOpportunities: {
+      reducer: (draftState, { payload }: { payload: GetOpportunityUserStakingDataOutput }) => {
+        draftState.userStaking.byId = merge(draftState.userStaking.byId, payload.byId)
+        draftState.userStaking.ids = Object.keys(draftState.userStaking.byId) as UserStakingId[]
```

Comment:
very smort, was going to comment we still need merge but we update the reference of the draftState above 🧠